pkg/kv/kvserver/logstore/logstore.go

```diff
@@ -563,7 +549,7 @@ func LoadTerm(
 	rsl StateLoader,
 	eng storage.Engine,
 	rangeID roachpb.RangeID,
-	eCache *raftentry.Cache,
+	eCache *raftentry.Cache, // TODO(pav-kv): this should be the caller's concern
```

Comment:
These 2 TODOs are tricky. There is a lot of intertwining between this code and the cache. I tried untangling and it's mostly doable, but [this](https://github.com/cockroachdb/cockroach/blob/fa5a923a582d2a668d80622eafd7178b684fa7c7/pkg/kv/kvserver/logstore/logstore.go#L593-L603) update resists.

Potentially this is a crux that wouldn't easily allow cleaner layers here. Maybe this PR should be held off for now.

Reply:
What makes this one update so difficult? As far as I can tell, both callers are ultimately a `*Replica`, and in particular `LoadTerm` is not part of any interface, and you seem to be able to change what's returned from it. We could thus add a return parameter for the entry to be cached, can't we?

Also, I agree with your comment that it is likely unhelpful/harmful to add to the cache here.

Reply:
#145559